### PR TITLE
Schedule Stuff

### PR DESCRIPTION
**Pull request by pcockrell6**
Added schedule stuff to the API

### CODE REVIEW

**Comment by zachsanchez113** [comment]
The file path is `routes/api/` for any API files, please move the files to that directory and I'll approve

**Comment by pcockrell6** [comment]
I think I fixed them. Put everything into the routes/api folder and recommitted to github

File: API/routes/api/get_schedule.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+			if (e) return next(e);
+
+			if (!query[0] || query[0] == "") return next(unregisteredError);
+
+			class = {
+
+				"startTime": query[0].startTime,
+				"endTime": query[0].endTime,
+				"dayOftheWeek": query[0].dayOftheWeek,
+				""
```
Comment (zachsanchez113) [inline API/routes/api/get_schedule.js:51]: Extraneous comma and quotes, I deleted them in the production code but this needs to be pushed at some point

File: API/routes/api/get_schedule.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+			}
+
+			console.log("Class fetch success for the user " + request.body.className + ".");
```
Comment (zachsanchez113) [inline API/routes/api/get_schedule.js:55]: Uh

File: API/routes/api/get_schedule.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+const router = express.Router();
+
+/*
+ *	
+ * Schedule API to be implemented.
+ * @name api/schedule
+ * 
+ */
+
+router.route('/')
```
Comment (zachsanchez113) [inline API/routes/api/get_schedule.js:22]: Documentation is needed

File: API/routes/api/make_schedule.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+ * @function
+ * @param {string} request.body.className - Name of the class
+ * @param {string} request.body.startTime - Starting time of the class
+ * @param {string} request.body.endTime - Ending time of the class
+ * @param {string} request.body.dayOftheWeek - Day of the week of the class
+ * @param {string} request.body.isOneTime - Whether the class is a one time event
+ * @param {string} request.body.classDate - Date that the class occurs [pending removal/reclassificaiton]
+ * @return {undefined} 200 if successful
+ * @return {undefined} 400 if a required field is missing
+ * @return {undefined} 400 if the email already exists in the system
```
Comment (zachsanchez113) [inline API/routes/api/make_schedule.js:28]: ?

File: API/routes/api/get_schedule.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+				(request.connection.socket ? request.connection.socket.remoteAddress : null);
+
+		ip = ip.replace('::ffff:','');
+
+		require('log-timestamp')(() => { return ip + '' + (new Date).toString().replace(' GMT-0500 (Eastern Standard Time', ' (US/New York)') });
+
+		console.log("Request to fetch schedule was received.");
+
+		/* @todo Check for class name and possibly other attributes */
+		if (!request.body.className) return next(missingClassNameError);
```
Comment (zachsanchez113) [inline API/routes/api/get_schedule.js:39]: Please use a `missingFieldError`, this error doesn't exist

File: API/routes/api/make_schedule.js [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+			console.log("Request to make a new schedule event was received.");
+
+			if (process.env.NODE_ENV == "debug") console.log("debug env variable success!");
+
+			/* Check for required fields */
+
+			if (!request.body.className || !request.body.startTime || !request.body.endTime) {
+				
+				console.error("Class creation attempt: One or more required fields is missing");
+				return response.status(400).send("One or more required fields is missing");
```
Comment (zachsanchez113) [inline API/routes/api/make_schedule.js:60]: If you could delete the `console.error` statement and replace the `return` with `return next(missingFieldError)`, that'd be appreciated. My error-handling middleware will log any errors if you use the `next()` function

File: API/routes/api/make_schedule.js [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+				}
+
+				/* Duplicate class name found if query result is non-empty or not null */
+
+				if (query[0]) {
+
+					console.error("New class creation failed: Class Name " + request.body.className + " already exists");
+					return response.status(400).send("Class Name already taken");
+
```
Comment (zachsanchez113) [inline API/routes/api/make_schedule.js:80]: Similar thing, except there's a way to specify a custom error. As an example from `files.js`:

`return next(createError(415, "The document must be one of the accepted types"));`

File: API/routes/api/get_schedule.js [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+		scheduleModel.find({"className": request.body.className}, 'startTime endTime dayOftheWeek', (e, query) => {
+			if (e) return next(e);
+
+			if (!query[0] || query[0] == "") return next(unregisteredError);
+
+			class = {
```
Comment (zachsanchez113) [inline API/routes/api/get_schedule.js:46]: Reserved keyword

Comment (zachsanchez113) [reply API/routes/api/get_schedule.js:46]: Looks like this was supposed to be `schedule`, fixed it

File: API/routes/api/get_schedule.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+		console.log("Request to fetch schedule was received.");
+
+		/* @todo Check for class name and possibly other attributes */
+		if (!request.body.className) return next(missingClassNameError);
+
+		scheduleModel.find({"className": request.body.className}, 'startTime endTime dayOftheWeek', (e, query) => {
+			if (e) return next(e);
+
+			if (!query[0] || query[0] == "") return next(unregisteredError);
```
Comment (zachsanchez113) [inline API/routes/api/get_schedule.js:44]: Incorrect error

File: API/routes/api/get_schedule.js [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+				"dayOftheWeek": query[0].dayOftheWeek,
+				""
+
+			}
+
+			console.log("Class fetch success for the user " + request.body.className + ".");
+			return response.contentType('application/json').status(200).send(JSON.stringify(schedule));
+		});
+
+	})
```
Comment (zachsanchez113) [inline API/routes/api/get_schedule.js:59]: Extraneous close, fixed it

**Comment by zachsanchez113** [comment]
Linked to issue #135 because I just did a bunch of spacing changes and this PR will obviously encompass that

**Comment by pcockrell6** [comment]
Pushed again with all the specific requested changes fixed

**Pull request merged**